diff --git a/include/fmt/core.h b/include/fmt/core.h
index f6a37af..ffabe63 100644
--- a/include/fmt/core.h
+++ b/include/fmt/core.h
@@ -286,7 +286,7 @@
 
 // Enable minimal optimizations for more compact code in debug mode.
 FMT_GCC_PRAGMA("GCC push_options")
-#if !defined(__OPTIMIZE__) && !defined(__NVCOMPILER)
+#if !defined(__OPTIMIZE__) && !defined(__NVCOMPILER) && !defined(__LCC__) && !defined(__CUDACC__)
 FMT_GCC_PRAGMA("GCC optimize(\"Og\")")
 #endif
 
